test/integration/test_konnect_extension.go

```diff
@@ -39,40 +51,325 @@ func TestKonnectExtensionKonnectGatewayControlPlaneNamespacedRef(t *testing.T) {
+	t.Logf("waiting for status.konnect and status.dataPlaneClientAuth to be set for KonnectExtension %s/%s", p.konnectExtension.Namespace, p.konnectExtension.Name)
+	require.EventuallyWithT(t,
+		checkKonnectExtensionStatus(p.konnectExtension, p.konnectControlPlane.GetKonnectID(), ""),
+		testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
 
-	t.Logf("Waiting for KonnectExtension %s/%s to have ControlPlaneRefValid condition set to True", ke.Namespace, ke.Name)
-	require.EventuallyWithT(t, func(t *assert.CollectT) {
-		err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: ke.Name, Namespace: ke.Namespace}, ke)
+	t.Logf("Creating a DataPlane using the KonnectExtension %s/%s", p.konnectExtension.Namespace, p.konnectExtension.Name)
+	dataPlane := builder.NewDataPlaneBuilder().
+		WithObjectMeta(metav1.ObjectMeta{
+			Namespace: p.namespace,
+			Name:      "test-konnect-extension",
+		}).
+		WithPodTemplateSpec(&corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  consts.DataPlaneProxyContainerName,
+						Image: helpers.GetDefaultDataPlaneEnterpriseImage(),
+						Env: []corev1.EnvVar{
+							{
+								Name:  "KONG_LOG_LEVEL",
+								Value: "debug",
+							},
+						},
+					},
+				},
+			},
+		}).
+		WithExtensions(
+			[]commonv1alpha1.ExtensionRef{
+				{
+					Group: konnectv1alpha1.GroupVersion.Group,
+					Kind:  "KonnectExtension",
+					NamespacedRef: commonv1alpha1.NamespacedRef{
+						Name: p.konnectExtension.Name,
+					},
+				},
+			},
+		).Build()
+	require.NoError(t, p.client.Create(ctx, dataPlane))
+	t.Cleanup(deleteObjectAndWaitForDeletionFn(t, dataPlane))
+
+	dpName := k8stypes.NamespacedName{
+		Namespace: dataPlane.Namespace,
+		Name:      dataPlane.Name,
+	}
+
+	t.Log("verifying dataplane gets marked provisioned")
+	require.Eventually(t, testutils.DataPlaneIsReady(t, GetCtx(), dpName, GetClients().OperatorClient), 10*waitTime, tickTime)
+
+	t.Logf("verifying dataplane %s has ingress service", dpName)
+	var dpIngressService corev1.Service
+	require.Eventually(t, testutils.DataPlaneHasActiveService(t, GetCtx(), dpName, &dpIngressService, clients, client.MatchingLabels{
+		consts.GatewayOperatorManagedByLabel: consts.DataPlaneManagedLabelValue,
+		consts.DataPlaneServiceTypeLabel:     string(consts.DataPlaneIngressServiceLabelValue),
+	}), waitTime, tickTime)
+
+	t.Log("verifying dataplane services receive IP addresses")
+	require.Eventually(t, func() bool {
+		err := p.client.Get(ctx, k8stypes.NamespacedName{
+			Namespace: dpIngressService.Namespace,
+			Name:      dpIngressService.Name,
+		}, &dpIngressService)
 		require.NoError(t, err)
-		assert.Truef(t, lo.ContainsBy(
-			ke.Status.Conditions, func(cond metav1.Condition) bool {
-				return cond.Type == konnectv1alpha1.ControlPlaneRefValidConditionType &&
-					cond.Status == metav1.ConditionTrue
+		return len(dpIngressService.Status.LoadBalancer.Ingress) > 0
+	}, 10*waitTime, tickTime)
```

Comment:
Do we need this 10x here?

```suggestion
	}, waitTime, tickTime)
```

Reply:
Nope, that was for testing purpose, great catch!